frontend/src/components/FundsPullPreApproval/FundsPullPreApproval.tsx

```diff
@@ -0,0 +1,83 @@
+import { Approval } from "../../interfaces/approval";
+import React from "react";
+import { classNames } from "../../utils/class-names";
+import { Button, Col, Container, Row } from "reactstrap";
+import ApprovalDetails from "./ApprovalDetails";
+
+interface ApprovalProps {
+  approval: Approval;
+  onApproveClick: () => void;
+  onRejectClick: () => void;
+  onRevokeClick: () => void;
+  onAnyClickSetApproval: () => void;
+  displayApproveRejectButtons: boolean;
+  displayRevokeButton: boolean;
+}
+
+function FundsPullPreApproval({
+  approval,
+  onApproveClick,
+  onRejectClick,
+  onRevokeClick,
+  onAnyClickSetApproval,
+  displayApproveRejectButtons,
+  displayRevokeButton,
+}: ApprovalProps) {
+  const itemStyles = {
+    "list-group-item": true,
+  };
+
+  const onAnyClick = (method: () => void) => () => {
+    method();
+    onAnyClickSetApproval();
+  };
+
+  return (
+    <li className={classNames(itemStyles)} key={approval.funds_pull_pre_approval_id}>
+      <Container>
+        <Row>
+          <Col sm="8" className="p-0">
+            <ApprovalDetails approval={approval} />
+          </Col>
+          {displayApproveRejectButtons && (
+            <Col sm="4" className="p-0 d-flex align-items-end">
+              <div className="mt-5 ml-auto">
+                <Button
+                  className="mr-1"
+                  color="black"
+                  outline
+                  size="sm"
+                  disabled={!onRejectClick}
+                  onClick={onAnyClick(onRejectClick)}
+                >
+                  <i className="fa fa-times mr-1" />
+                  Reject
+                </Button>
+                <Button
+                  color="black"
+                  size="sm"
+                  disabled={!onApproveClick}
+                  onClick={onAnyClick(onApproveClick)}
+                >
+                  <i className="fa fa-check mr-1" />
+                  Approve
+                </Button>
+              </div>
+            </Col>
+          )}
+          {displayRevokeButton && (
+            <Col sm="4" className="p-0 d-flex align-items-end">
+              <div className="mt-5 ml-auto">
+                <a href="#" onClick={onAnyClick(onRevokeClick)} aria-disabled={!onRevokeClick}>
```

Comment:
Avoid using `a` tags in React without intention to actually redirect the user outside of React app. You probably need a `button` here. If you need it to look like a link you can use the class `btn-link`

Reply:
![Screenshot from 2021-02-08 15-33-37](https://user-images.githubusercontent.com/11321771/107226694-1bef4f80-6a23-11eb-8059-df48d1cc1196.png)

@vladkucherov - How I get rid the borders?

Reply:
`border-0`

Wondering why there is a border for `btn-link`. By default I think shouldn't be